src/realm/object-store/shared_realm.cpp

```diff
@@ -912,6 +913,10 @@ void Realm::close()
 
 void Realm::delete_files(const std::string& realm_file_path)
 {
+    if (!realm::util::File::exists(realm_file_path)) {
+        return;
+    }
+
```

Comment:
It is problematic that the code below relies on ordering inside the core_files... what if a folder occurs there before the files inside it? Will it then fail to delete the folder since it is not empty?

Reply:
@finnschiermer  Not sure I understand. All files and folders are on the same level: where the Realm file is.
The only folder here is the `.management` folder but the `core_files` do not include any files inside it but just the folder itself.

Reply:
The `core_files` do not have an order since they are just a `std::unordered_map<DB::CoreFileType, std::pair<std::string, bool>>`.

Reply:
got it.